### PR TITLE
itvr-296-304-312--bcsc--dynamic-keycloak

### PR DESCRIPTION
**Pull request by tim738745**
(1) This replaces PR #127 (now closed); it appears that having multiple keycloaks initialized at the same time would cause the app to break (intermittently when only one keycloak was logged in, and completely when two keycloaks were logged in). Also, on Safari, the app would get into this infinite redirect loop immediately upon landing on the home page.

(2) Now, only one keycloak is initialized at a time, depending on what the user selects. The user's selection is persisted through redirects/refreshes using localStorage. Also, the app does not break if the user manages to log into both bceid and bcsc at the same time.

(3) I'm still not sure if this solution is ideal; what would be ideal is if the bceid and bcsc idps are both in the same keycloak realm; if this is not possible from a security standpoint, this video (https://www.youtube.com/watch?v=lJlKNmAsaiE) suggests using subdomains to determine which keycloak to use. For example, we would have bcsc.goelectric.ca, and bceid.goelectric.ca. However, this would involve UI redesigns and possibly need to coordinate with the people designing the "go electric" website that will point to our app...

### CODE REVIEW

File: keycloak.env [hunk 1]
```diff
@@ -1,3 +1,7 @@
-KEYCLOAK_CLIENT_ID=itvr-2674
-KEYCLOAK_REALM=onestopauth-basic
-KEYCLOAK_URL=https://dev.oidc.gov.bc.ca/auth/
+BCEID_KEYCLOAK_CLIENT_ID=itvr-2674
+BCEID_KEYCLOAK_REALM=onestopauth-basic
+BCEID_KEYCLOAK_URL=https://dev.oidc.gov.bc.ca/auth/
+
+BCSC_KEYCLOAK_CLIENT_ID=zeva
```
Comment (naomiaro) [inline keycloak.env:5]: This should still have a separate itvr client setup.

File: frontend/src/index.js [hunk 2]
```diff
@@ -2,23 +2,23 @@ import React from 'react';
...
-const initOptions = {
-  onLoad: 'check-sso',
-  pkceMethod: 'S256',
-  useNonce: false
-};
+let keycloak = keycloaks[Object.keys(keycloaks)[0]];
+const realm = localStorage.getItem('keycloakRealm');
+if (realm) {
+  keycloak = keycloaks[realm];
+}
```
Comment (naomiaro) [inline frontend/src/index.js:16]: maybe this keycloak check can be moved into `./keycloak`?

**Review by naomiaro — CHANGES_REQUESTED** [review]
Hopefully the two keycloak realms will just be temporary for a few months and bcsc is in the common hosted realm soon. Besides just setting up a separate client for itvr, and then giving a list of env vars to Kuan, lgtm 😄

**Comment by tim738745** [comment]
@naomiaro I made the changes, setup a new itvr client in the rzh2zkjq realm, and gave Kuan the new env vars; should be good now!

**Review by naomiaro — APPROVED** [review]

**Pull request merged**